src/main/java/com/cwjoshuak/CamTorumMiningPlugin.java

```diff
@@ -142,6 +148,51 @@ private void onTileObject(Tile tile, TileObject oldObject, TileObject newObject)
 		}
 	}
 
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (!inCamTorumMiningArea || streams.isEmpty() || !config.notifyWater())
+		{
+			return;
+		}
+
+		int ticksSinceNotif = client.getTickCount() - lastNotificationTick;
+		if (ticksSinceNotif < 52)
+		{ // streams last for about 45 or 50 game ticks
+			return; // already notifier for current set
+		}
+
+		lastNotificationTick = client.getTickCount();
+
+		boolean alreadyMiningStream = false;
+		WorldPoint wp = client.getLocalPlayer().getWorldLocation();
+		for (Map.Entry<TileObject, Tile> entry : streams.entrySet())
+		{
+			Tile tile = entry.getValue();
+			if (tile.getWorldLocation().distanceTo(client.getLocalPlayer().getWorldLocation()) >= config.maxDistance())
+			{
+				continue;
+			}
+
+			int dist = Math.abs(wp.getX() - tile.getWorldLocation().getX()) + Math.abs(wp.getY() - tile.getWorldLocation().getY());
+			if (dist != 1)
+			{ // manhattan distance of 1 is adjacent and not diagonal to a stream-the tile to be able to mine it
+				continue;
+			}
+
+			if (client.getLocalPlayer().getAnimation() >= 0)
```

Comment:
I think we can remove this, because this causes it to NOT notify if you are mining a non-stream.

I think I'd like to receive a notification for when it spawns, if you are not already mining a stream, which the code above this line handles.